pkg/provider/cloud/vmwareclouddirector/provider.go

```diff
@@ -357,3 +357,36 @@ func ListOVDCNetworks(ctx context.Context, auth Auth) (apiv1.VMwareCloudDirector
 
 	return orgVDCNetworks, nil
 }
+
+func ListStorageProfiles(ctx context.Context, auth Auth) (apiv1.VMwareCloudDirectorStorageProfileList, error) {
+	client, err := NewClientWithAuth(auth)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create VMware Cloud Director client: %w", err)
+	}
+
+	org, err := client.GetOrganization()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get organization %s: %w", auth.Organization, err)
+	}
+
+	orgVDC, err := client.GetVDCForOrg(*org)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get organization VDC '%s': %w", auth.VDC, err)
+	}
+
+	var storageProfiles apiv1.VMwareCloudDirectorStorageProfileList
+
+	if orgVDC.Vdc.VdcStorageProfiles == nil {
+		return storageProfiles, fmt.Errorf("no storage profiles exist for %s VDC", auth.VDC)
+	}
```

Comment:
why is this an error? Could we return just the empty storage profile list? Or is this something so unexpected that it deserves an error?

Reply:
It makes sense to just return an empty list 👍